### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by moham96** [tobloef/deep-utility-types#4]
If the type is deep enough, typescript will lag and throw this error "Type instantiation is excessively deep and possibly infinite"

**Comment by tobloef** [comment]
I suppose a solution could be to add a maximum depth to the recursive types, which should be fairly doable.

I'm open to a PR, but I'm probably not going to have time to fix this issue myself at the moment.

**Comment by d9k** [comment]
Can we just add an option to limit level?

**Comment by kakasoo** [comment]
@moham96 @tobloef 

If you don't mind, I made a type to solve the above problem, is it okay if you link it here? My open source is similar to your type, but I'm dealing with your issues more deeply. If you don't have time to develop it, how about sharing my open source, and developing it together later? I want someone to develop with me. If what I'm saying is weird it's because my main language is not English, so please understand.

**Comment by tobloef** [comment]
@kakasoo Yes, feel free to share a link to your repository if you have something others might find relevant 😃

**Comment by kakasoo** [comment]
Thanks!

I've created types like `DeepStrictObjectKeys`, `DeepStrictPick`, and `DeepStrictOmit` that can infer keys regardless of depth, and now I want to provide them in the form of functions one by one. Please use them if you're interested.

- [npm @kakasoo/deep-strict-types](https://www.npmjs.com/package/@kakasoo/deep-strict-types)
- [github repository](https://github.com/kakasoo/deepstricttypes)
